### PR TITLE
Email features generation

### PR DESCRIPTION
**Pull request by rioman**
Generate email features with given in constructor domain.

### CODE REVIEW

**Review by Ijka — APPROVED** [review]

File: src/syngen/ml/vae/models/features.py [hunk 1]
```diff
@@ -572,6 +572,32 @@ def loss(self) -> tf.Tensor:
...
+        domain: str = 'syngenai.com'
+    ):
+        super().__init__(name=name,
+                         text_max_len=text_max_len,
+                         rnn_units=rnn_units,
+                         dropout=dropout)
+        self.domain = domain
+
+    def transform(self, data: pd.DataFrame) -> np.ndarray:
+        if len(data.columns) > 1:
```
[the remark this answers is not in the document]
Comment (rioman) [reply src/syngen/ml/vae/models/features.py:591]: It's necessary here for I pass only first column of dataset to inherited method.

File: src/syngen/ml/vae/models/dataset.py [hunk 2]
```diff
@@ -456,7 +462,8 @@ def _set_long_text_columns(self, df: pd.DataFrame):
         """
         Set up the list of columns with long texts (> 200 symbols)
         """
-        data_subset = self._select_str_columns(df)
+        text_columns = self._select_str_columns(df)
+        data_subset = df[text_columns]
 
         self.long_text_columns = set()
```
Comment (rioman) [inline src/syngen/ml/vae/models/dataset.py:468]: Shouldn't this line be deleted?
See also line 393

**Review by Anna050689 — APPROVED** [review]

**Review by Anna050689 — CHANGES_REQUESTED** [review]
It looks that the regex expression should be modified because it accepts the invalid email that can start from "." sign
I think that this expression might be more precise - 
**r"([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+"**

It will be modified related to the this requirements - https://en.wikipedia.org/wiki/Email_address#Standards_documents

File: src/syngen/ml/vae/models/dataset.py [hunk 3]
```diff
@@ -864,7 +913,7 @@ def _preprocess_nan_cols(
                 fillna_value = self.df[feature].mean()
             elif fillna_strategy == "mode":
                 fillna_value = self.df[feature].dropna().mode().sample(1).values[0]
-            elif fillna_strategy == "text":
+            elif (fillna_strategy == "text") or (fillna_strategy == "email"):
```
Comment (Anna050689) [inline src/syngen/ml/vae/models/dataset.py:916]: I think that it's not necessary to add new fill_strategy - "email"

Comment (rioman) [reply src/syngen/ml/vae/models/dataset.py:916]: fillna_strategy takes part in formulating of log records (line 910), so it should be separate for email.

**Pull request merged**